admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/SkuCustomPersistenceHandler.java

```diff
@@ -741,13 +740,16 @@ public Entity update(PersistencePackage persistencePackage, DynamicEntityDao dyn
                 return entity;
             }
 
+            // Persist our updates before making any other modifications.  Updating/removing
+            // ProductOptionValues cause an `em.flush()` downstream which can cause our changes
+            // to not be detected by the workflow management.
+            adminInstance = dynamicEntityDao.merge(adminInstance);
```

Comment:
Curious why this is necessary. WorkflowEntityPersisterUpdateInterceptorExtensionHandler catches every flush and includes stuff in a ChangeDetailContext, which is eventually merged in SimpleChangeDetailDaoImpl. This should be capable of catching any persistence interaction(s) with the database during this flow and recording the change details.

Reply:
I believe it has to do with `AdminSandBoxEntityManagerInvocationProvider#handleRemoveState` which is hit when the `ProductOptionValues` are removed from the sku.  When the `WorkflowEntityPersisterUpdateInterceptorExtensionHandler` was being hit for the sku, it had `WorkflowStateManagerImpl.getWorkflowStateManager().getIgnoreUpdateInsertState()` set to true which caused it to skip the code that populates the `changesByTable` map.

Before swapping the order, `WorkflowEntityPersisterUpdateInterceptorExtensionHandler` was never hit when the merge was called for sku if it had product option values associated with it.  I'm guessing that the initial time through was because of a cascade from the remove but I'm not entirely sure.

Reply:
I don't think one-off ordering changes are the right spot to try to deal with an idiosyncrasy of  WorkflowEntityPersisterUpdateInterceptorExtensionHandler. The design intent is that whatever is making it to the database during a flush (implicit or explicit) is interpreted by WorkflowEntityPersisterUpdateInterceptorExtensionHandler. It sounds like getIgnoreUpdateInsertState() caused that to fail in this case? If we try to think about this as messing with ordering in various spots, I don't think we'll ever get it right, especially across both strategies.

Reply:
It may be that everything dirty was flushed during the product option value maintenance, and there was nothing to persist at the final merge.

Reply:
Taking a quick look at the code in WorkflowEntityPersisterUpdateInterceptorExtensionHandler, I don't think the setupDetails methods should be gated with `!WorkflowStateManagerImpl.getWorkflowStateManager().getIgnoreUpdateInsertState()`.

Reply:
That's what I was seeing.  I actually just pulled that out of the if statement and was going to run some tests to see what it affected.

Reply:
@jefffischer I'm not seeing any immediate issues with removing the gating.  I'll close this PR and continue testing that solution.